### ISSUE TITLE
Unexpected static keyword

### ISSUE THREAD

**Issue by zizi4n5** [nicklockwood/SwiftFormat#1498]
With this code:

```test.swift
@_silgen_name("foo")
func foo(_: CFString, _: CFTypeRef) -> Int?

enum Bar {
    static func baz() {
        fatalError()
    }
}
```

Running `swiftformat test.swift --self init-only` fails with:

```bash
error: Unexpected static keyword on line 5 in test.swift.
```

using 0.51.15
Up to 0.51.10 of SwiftFormat, the above code did not cause an error.
Starting with commit(https://github.com/nicklockwood/SwiftFormat/commit/3c4d2883cf36c03cf84b468c836b6d934feb28ef) added in 0.51.11, the error occurs.

@_silgen_name:
https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#using-_silgen_name-to-call-c-from-swift

**Comment by nicklockwood** [comment]
@zizi4n5 thanks for doing the work to track this down, that was a big help! It'll be fixed in the next release.

**Comment by nicklockwood** [comment]
@zizi4n5 fixed in 0.52.0
